src/util/stack_overflow_detection.cpp

```diff
@@ -0,0 +1,115 @@
+//  Copyright (c) 2017 Abhimanyu Rawat
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// for Visual C++ compiler
```

Comment:
@ABresting This is confusing. The comment explains that this file is supposed to be included in builds with MSVC, but the conditional inclusion is doing exactly the opposite. And the build is failing because of a missing `config.h` header.